src/classes/application.js

```diff
@@ -12,22 +13,35 @@ module.exports = class Application extends Base {
     const config = require(path.join(process.cwd(), 'config/environment.js'))(process.env.NODE_ENV)
     const MiddlewareRouter = require(path.join(process.cwd(), 'app/middleware.js'))
     const Router = require(path.join(process.cwd(), 'app/router.js'))
+    const log = new Log()
 
+    log.trace('Ash server creating express app instance')
     const app = express()
     _app.set(this, app)
 
-    const log = new Log()
-
-    log.info('Ash server loading middleware')
+    log.trace('Ash server loading middleware')
     const middlewareDir = path.join(process.cwd(), 'app/middleware')
     loadMiddleware(MiddlewareRouter.definition, app, middlewareDir)
 
-    log.info('Ash server loading routes')
+    log.trace('Ash server loading routes')
     const options = {routesDir: path.join(process.cwd(), 'app/routes')}
     app.use(createRoutes(Router.definition, options))
 
+    const initializerDir = path.join(process.cwd(), 'app/initializers')
```

Comment:
> How does ember allow importing from the app base?

Brocoli/ember-cli-build. It's a bit different though, here, isn't it, because we want to avoid a precompile step.

I'm hesitant to suggest this, but maybe one way to do it is to have a naive kind of build/precompile step where we build out a `/build` directory or something and selectively `ln -s` files to the correct location. Otherwise, maybe through dynamic imports or something, though I don't know how that would work unless we provided an alternative to `require` that we can hook (e.g. `requireAsh('app/some/module')` figures out which file to import), similar to how electron does it.

Actually, now that I say that, we can leverage the "convention over configuration" rule quite a bit to our favor if necessary. By using convention everywhere (meaning the framework knows which files to import and when), the user doesn't need to explicitly import things.

Example

```js
// Normal node way
const Authentication = require('app/services/authentication')
class Route extends Ash.Route {
  authentication: Authentication()
}
```

In the first example, it's hard to do any file-tree merging, because it's an import to an explicit location.

```js
// Convention way
const Ash = require('ash-framework')
class Route extends Ash.Route {
  authentication: Ash.service.inject()
}
```

In the second example, we delegate the "import" to the framework, which allows it to select which file to import and create the Authentication service from. In other words, the user "registers"  instead of `require`s.

```js
if (fileExists('app/services/authentication')) {
  return require('app/services/authentication')
} else {
  for (addon in registeredAddons) {
    if (fileExists(`${addon.rootDir}/app/services/authentication`) {
      return require(`${addon.rootDir}/app/services/authentication`)
    }
  }
}

if (fileExists('@ash-framework/app/services/authentication') {
  return require('@ash-framework/app/services/authentication')
}
```